platform-tooling-support-tests/src/main/java/platform/tooling/support/Helper.java

```diff
@@ -115,17 +115,8 @@ static JarFile createJarFile(String module) {
 			() -> System.getenv("JAVA_HOME_" + version), //
 			() -> System.getenv("JAVA_" + version) //
 		);
-		var home = sources.stream().map(Supplier::get).filter(Objects::nonNull).findFirst();
-		if (home.isPresent()) {
-			return Optional.of(Path.of(home.get()));
-		}
-		// Try to inspect Maven Toolchains configuration file...
-		var jdkHome = getJdkHomeFromMavenToolchains(version);
-		if (jdkHome.isPresent()) {
-			return jdkHome;
-		}
-		// Still here? Return an empty optional.
-		return Optional.empty();
+		var homePath = sources.stream().map(Supplier::get).filter(Objects::nonNull).findFirst().map(Path::of);
+		return homePath.or(() -> getJdkHomeFromMavenToolchains(version));
```

Comment:
I'm reverting this change.

Rationale: the original author of this code chose to split up the steps and add comments in order to simplify understanding of the steps involved.

Reply:
I found it pretty unidiomatic use of the optional api, so to me at least this procedural style with all the split out steps made it less readable. As for the comments - in this case I didn't think the comments were adding anything - they are telling the same story as the code, which is a good place to be in. I can see where you're coming from though.